packages/components/src/placeholder/index.tsx

```diff
@@ -76,9 +78,19 @@ export function Placeholder< IconProps = unknown >(
 		modifierClassNames,
 		withIllustration ? 'has-illustration' : null
 	);
+
 	const fieldsetClasses = classnames( 'components-placeholder__fieldset', {
 		'is-column-layout': isColumnLayout,
 	} );
+
+	const debouncedSpeak = useDebounce( speak, 100 );
+
+	useEffect( () => {
+		if ( instructions ) {
+			debouncedSpeak( instructions );
+		}
+	}, [ instructions, debouncedSpeak ] );
```

Comment:
Should this run every time the block is selected? As of now, the instructions are only provided on insert.

```suggestion
	}, [] );
```

Reply:
Well, the pre-commit linter forced me to add both props as dependencies. I would have gone with an empty array otherwise.

Reply:
With an empty dependency array, if the instructions were to be updated, they would not be announced again — in that sense, having both dependencies looks correct to me?

Regarding `debouncedSpeak`, the `useDebounce` hook [internally uses `useMemo`](https://github.com/WordPress/gutenberg/blob/6692753fb9894b41dd75f8e21dff1f4da164914f/packages/compose/src/hooks/use-debounce/index.js#L32), and therefore the `debounceSpeak` dependency should not really change unless the `speak` function changes

Reply:
`instructions` doesn't change as well, at least there are no cases or dynamic instructions so far. So, regardless f whether we use an empty array or not, the announcement is not repeated. Anyways, the pre-commit linter check requires to add both dependencies :) can't commit without adding them.

Reply:
@ciampo Is there anything we can use in place of `useDebounce()` to still have the timeout for the `speak()` but can announce on every render? I guess we could attach `onFocus` callback and trigger the announcement that way?

Reply:
Just a question: are we really sure we do want the announcement on every render? Personally, I'd find it a bit annoying. As a user, when I add a block I can hear the instructions once. That's enough for me, I wouldn't need to hear that again and again. It could also be unexpected. For example, in the Image block: pressing the 'Insert from URL' button makes a small popover appear where users can type an image URL. That triggers a render and the instructions are announced again, in the middle of a user flow where they would be totally unexpected. I'm afraid there's no way to predict what _kind_ of re-render actually occurs. In some cases, making the instructions be announced again could be of some value. Not sure that would be appropriate for any re-render though.

Reply:
@afercia It is a fair question and most users probably do not need this. However, if you go back to edit a post later and it has been some time since you interacted with the block, a reminder on how to interact with it may be needed. With no good way to access the placeholder text itself, I would vote for slightly more info than not enough even though verbosity is a very real issue in dynamic web apps today.

Reply:
It sounds like the first step here is to understand the desired behavior — should the announcement be heard or every render? Only when the block is added? Every time the block is focused? Every time the block is focused?

Once there is an agreement there, we can think about what's the best way to implement the desired behaviour

Reply:
Ideally we should hear it when the block is focused, because that way when adding placeholders, saving, then going back some time later to add the actual content, the user will know how to interact with the block. 

Also worth considering the use case of interacting with an existing template where all the blocks are already in place; the person that fills in the content might not even be the same as the one who adds the blocks.

If the problem with the fieldset/legend combo is that not all placeholders have controls, I wonder if we could come up with a different markup structure for those, and some sort of flag to let the component know when to use either? This has probably been suggested already but nothing else occurs to me 😅

Reply:
@tellthemachines I think what we really need is to tie the `Placeholder` and block selection together. That way the announcement is triggered on block select. I am just not sure how to do this with different components at play. Does the `Placeholder` render if the block is not selected?

Reply:
`Placeholder` renders whenever a new block is inserted or, after save and reload, if the block hasn't been changed since insertion. It'll render whether the block is focused or not, because it's always meant to be visually present on the page.

Reply:
@tellthemachines Think it is possible to add `onFocus` callback to the wrapping div and just call the `speak()` function when focus is actually placed inside the block content?